src/Parsers/ParserAlterQuery.cpp

```diff
@@ -725,8 +731,24 @@ bool ParserAlterCommand::parseImpl(Pos & pos, ASTPtr & node, Expected & expected
                 if (!parser_modify_col_decl.parse(pos, command_col_decl, expected))
                     return false;
 
+                auto check_no_type = [&](const std::string_view keyword)
+                {
+                    if (!command_col_decl)
```

Comment:
Can this happen? It should have returned in https://github.com/ClickHouse/ClickHouse/pull/68935/files#diff-894d69924e6477d08b80e48da50817ca27b597b2a13de72de094667dcfff7fbcR731